src/App.tsx

```diff
@@ -1,8 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { getGoods } from './api/api';
+import { Product } from './interfaces/Product';
+import { GoodsList } from './components/GoodsList';
+import { Button } from './components/Button';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const load = () => {
+    setIsLoading(true);
+
+    return getGoods().then(data => {
+      setIsLoading(false);
+
+      return data;
+    });
+  };
+
+  const loadAll = (): void => {
+    load().then((data: Product[]) => setGoods(data));
+  };
+
+  const loadPart = (): void => {
+    load().then((data: Product[]) => {
+      const goodsList = data
+        .sort((a: Product, b: Product) => a.name.length - b.name.length);
```

Comment:
обрыв строки
опиши что возвращает функция load, тогда тут описывать не нужно
```suggestion
      const goodsList = data.sort((a, b) => (
        a.name.length - b.name.length
      ));
```